tofhir-engine/src/main/scala/io/tofhir/engine/mapping/job/FhirMappingJobManager.scala

```diff
@@ -348,21 +348,29 @@ class FhirMappingJobManager(
   /**
    * Read and join the source data
    *
-   * @param task           FHIR Mapping task
-   * @param sourceSettings The source settings of the mapping job
-   * @param timeRange      Time range for the source data to load
-   * @param jobId          The identifier of mapping job which executes the mapping
+   * @param task            FHIR Mapping task
+   * @param sourceSettings  The source settings of the mapping job
+   * @param timeRange       Time range for the source data to load
+   * @param jobId           The identifier of mapping job which executes the mapping
+   * @param isTestExecution Indicates whether the execution is a test
    */
   def readJoinSourceData(task: FhirMappingTask,
                          sourceSettings: Map[String, MappingJobSourceSettings],
                          timeRange: Option[(LocalDateTime, LocalDateTime)] = None,
-                         jobId: Option[String] = None): (FhirMapping, MappingJobSourceSettings, DataFrame) = {
+                         jobId: Option[String] = None,
+                         isTestExecution: Boolean = false): (FhirMapping, MappingJobSourceSettings, DataFrame) = {
     // if the FhirMapping task includes the mapping to be executed (the case where the mapping is being tested), use it,
     // otherwise retrieve it from the repository
     val mapping = task.mapping match {
       case Some(mapping) => mapping
       case None => fhirMappingRepository.getFhirMappingByUrl(task.mappingRef)
     }
+
+    // Check whether the mapping is draft and the execution is not a test.
```

Comment:
Let's change it to:

// ensure that the mapping is not marked as draft unless this is a test execution.